5/tests/unit/GraphQLBackendWithNeo4J.spec.js

```diff
@@ -0,0 +1,312 @@
+        await query({
+            query: ADD_DEPENDENCY,
+            variables: { id: 0, dependencyId: 1 },
+        })
+        await query({
+            query: ADD_DEPENDENCY,
+            variables: { id: 0, dependencyId: 2 },
+        })
+
+        const res = await query({
+            query: GET_DEPENDENCIES,
+            variables: { id: 0 },
+        })
+
+        expect(res.data.getToDoDependencies.length).toEqual(2)
+    })
+
+    it('gets all todos in the DB ordered by ascending IDs', async () => {
+        const driver = createNewDriver()
+        const server = createNewServer(driver)
+        const { query } = createTestClient(server)
+        let testUser = createJwt('secret')
+
+        await addMultipleTodos(25, query, testUser)
+
+        const res = await query({
+            query: GET_TODOS,
+            variables: {
+                FILTER_MODE: FILTER_MODE.ASC,
+                skip: -1,
+                limit: -1,
+            },
+        })
+        let todos = res.data.todos
+
+        for (let i = 0; i < todos.length - 1; i++) {
+            expect(todos[i].id <= todos[i + 1].id).toEqual(true)
+        }
+    })
+
+    it('gets all todos in the DB ordered by ascending IDs with paging', async () => {
+        const driver = createNewDriver()
+        const server = createNewServer(driver)
+        const { query } = createTestClient(server)
+        let testUser = createJwt('secret')
+
+        await addMultipleTodos(25, query, testUser)
+
+        const res = await query({
+            query: GET_TODOS,
+            variables: {
+                FILTER_MODE: FILTER_MODE.ASC,
+                skip: 1,
+                limit: 10,
+            },
+        })
+        let todos = res.data.todos
+
+        expect(todos.length).toEqual(10)
+
+        for (let i = 0; i < todos.length - 1; i++) {
+            expect(todos[i].id <= todos[i + 1].id).toEqual(true)
+        }
+    })
+
+    it('deletes a todo', async () => {
+        const driver = createNewDriver()
+        const server = createNewServer(driver)
+        const { query } = createTestClient(server)
+        const testUser = createJwt('secret')
+
+        let toBeDeleted = [
+            { id: 9999, newMessage: 'newentry', userAuth: testUser, lastEdited: new Date().toLocaleString() },
+        ]
+
+        await query({
+            query: ADD_TODO,
+            variables: toBeDeleted[0],
+        })
+
+        await query({
+            query: DELETE_TODO,
+            variables: {
+                id: 9999,
+                userAuth: testUser,
+            },
+        })
+
+        const res = await query({
+            query: GET_TODOS,
+            variables: {
+                FILTER_MODE: FILTER_MODE.NONE,
+                skip: -1,
+                limit: -1,
+            },
+        })
+
+        //https://jestjs.io/docs/en/expect#expectarraycontainingarray
+        expect(res.data.todos).not.toEqual(expect.arrayContaining(toBeDeleted))
```

Comment:
```suggestion
        await expect(
            query({
                query: GET_TODOS,
                variables: {
                    FILTER_MODE: FILTER_MODE.NONE,
                    skip: -1,
                    limit: -1,
                },
            })
        ).resolves.toMatchObject({
            data: {
                todos: toBeDeleted
            },
            errors: undefined
        });
```
You can compare the results like this and get much better Results if something went wrong during the test.